tests/lib/formatters/visualstudio.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * @fileoverview Tests for VisualStudio format.
+ * @author Ronald Pijnacker
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var assert = require("chai").assert,
+    formatter = require("../../../lib/formatters/visualstudio");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+describe("formatter:visualstudio", function() {
+    describe("when passed no messages", function() {
+        var code = [{
+            filePath: "foo.js",
+            messages: []
+        }];
+
+        it("should return nothing", function() {
+            var result = formatter(code);
+            assert.equal(result, "");
+        });
+    });
+
+    describe("when passed a single message", function() {
+        var code = [{
+            filePath: "foo.js",
+            messages: [{
+                message: "Unexpected foo.",
+                severity: 2,
+                line: 5,
+                column: 10,
+                ruleId: "foo"
+            }]
+        }];
+
+        it("should return a string in the format filename(x,y):\t[Error]   z for errors", function() {
+            var result = formatter(code);
+            assert.equal(result, "foo.js(5,10):\t[Error]   Unexpected foo.\t(foo)\n\n1 problem");
```

Comment:
Seems like the format isn't quite right. According to the docs here's what it should look like:

```
C:\sourcefile.cpp(134) : error C2143: syntax error : missing ';' before '}'
```

So I'm expecting the output to look more like this:

```
foo.js(5,10): error foo: Unexpected foo
```

Also, is it safe to have the "1 problem" at the end?